### ISSUE TITLE
NETLINK_SOCK_DIAG support?

### ISSUE THREAD

**Issue by est** [svinota/pyroute2#291]
I wanna use this library to build my own `ss` command utility from iproute2 suite.

Is it possible to include `NETLINK_SOCK_DIAG` in pyroute2?

Thanks!

**Comment by est** [comment]
btw 

http://www.spinics.net/lists/linux-man/msg10393.html

> As NETLINK_INET_DIAG should not longer be used, NETLINK_SOCK_DIAG should be
written instead.

Should we update pyroute2/netlink/__init__.py definitions?

**Comment by svinota** [comment]
Yes, I hope it is possible :) Right now I'm a bit busy porting the compat code to the master branch, so almost all the tickets are on «hold», but I hope to finish the porting this weekend, so we'll see.

**Comment by svinota** [comment]
Yep, it is possible. Give me some time to make a proof-of-concept, I hope it'll be ready the next week.

**Comment by svinota** [comment]
The initial commit. I managed to get the AF_UNIX socket info response (see comments in the code). More functions will follow the next week, but you can already start to play with the module.

**Comment by svinota** [comment]
`get_sock_stats()` is working (but only `unix_diag_req/unix_diag_msg` are implemented yet)

You're welcome to try it out. Tcp diag structures will follow soon — as I said, the next week.

**Comment by Pen-cell** [comment]
Can I assign good friends to handle this 
- Or non profit

**Comment by DavidAntliff** [comment]
> Tcp diag structures will follow soon — as I said, the next week.

Just curious - did `tcp_diag` support via `NETLINK_SOCK_DIAG` get into pyroute2 in the end?

**Comment by svinota** [comment]
@DavidAntliff you can take a quick look at the current status here: https://github.com/svinota/pyroute2/blob/master/pyroute2/netlink/diag/__init__.py

**Comment by DavidAntliff** [comment]
@svinota that looks promising - thank you.
